packages/api/src/useAPI.ts

```diff
@@ -12,6 +12,13 @@ import APIContext from './APIContext';
 
 const useAPI = () => {
     const api = useContext(APIContext);
+    const [isReady, setIsReady] = useState<boolean>(api?.get()?.connection?.readyState === WebSocket.OPEN);
+
+    useEffect(() => {
+        if ((api?.get()?.connection?.readyState === WebSocket.OPEN) !== isReady) {
+            setIsReady(api?.get()?.connection?.readyState === 1);
+        }
+    }, [api, isReady]);
```

Comment:
The websocket might open/close via an event. This will only change when the api variable is reassigned.

`addEventListener('open', ...`

Reply:
@markwylde-deriv good point. I had doubts about React flow here, thank you for your comment, I'll change it

Reply:
@markwylde-deriv looks like we really need to check `readyState`, not just `open/close`, because state can be `Connecting/Open/Closing/Closed`. I've tried to add `addEventListener` for open and close events and when I switch real/demo accounts I see logout issue with this implementation. Made decision to create one more const `readyState` to trigger `useEffect` by it.